src/meta/protos/proto/config.proto

```diff
@@ -37,3 +37,13 @@ message FsStorageConfig {
 
   string root = 1;
 }
+
+message GcsStorageConfig {
+  uint64 version = 100;
+  uint64 min_compatible = 101;
+
+  string endpoint_url = 1;
+  string bucket = 2;
+  string root = 3;
+  string credential = 4;
+}
```

Comment:
Every time a protobuf changes, increase `VER`, e.g. `VER=4`:
https://github.com/datafuselabs/databend/blob/2dfb0a2b3236cc861e81633cf67dcb94e3d25eec/src/meta/proto-conv/src/util.rs#L17

And add a test that deserializes `VER=4` messages. Mark the message for this test with a version suffix. Similar to:
https://github.com/datafuselabs/databend/blob/2dfb0a2b3236cc861e81633cf67dcb94e3d25eec/src/meta/proto-conv/tests/it/user_proto_conv.rs#L198-L210

Such a test is also required for the modified message `StageStorage`.

These tests ensure future version databend-query will deserialize a message of `VER=4` smoothly until `MIN_COMPATIBLE_VER` becomes higher than `4`, which means these formats do not need to be compatible anymore.

Reply:
Cc @ClSlaid, please help address this issue.

Reply:
Sorry for the late reply, I've been busy with some family issues, working in progress now.